diff --git a/dist/eslint.mjs b/dist/eslint.mjs
index 03e90f6a8d4487518cf0f0905cf756649f858c31..d92a041a9b9af3546146e1dfcda50129a5915b53 100644
--- a/dist/eslint.mjs
+++ b/dist/eslint.mjs
@@ -37,7 +37,7 @@ const markdownProcessor = {
     const links = extractMarkdownLinks(text);
     if (!links.length) {
       lineMapStack.push(/* @__PURE__ */ new Map());
-      return [{ text, filename: "0.md" }];
+      return [];
     }
     const virtualLines = [];
     const lineMap = /* @__PURE__ */ new Map();
